ext/sodium/libsodium.c

```diff
@@ -0,0 +1,2682 @@
+	PHP_FE(sodium_crypto_box_open, AI_StringAndNonceAndKey)
+	PHP_FE(sodium_crypto_box_publickey, AI_Key)
+	PHP_FE(sodium_crypto_box_publickey_from_secretkey, AI_Key)
+#ifdef crypto_box_SEALBYTES
+	PHP_FE(sodium_crypto_box_seal, AI_StringAndKey)
+	PHP_FE(sodium_crypto_box_seal_open, AI_StringAndKey)
+#endif
+	PHP_FE(sodium_crypto_box_secretkey, AI_Key)
+	PHP_FE(sodium_crypto_kx, AI_FourStrings)
+	PHP_FE(sodium_crypto_generichash, AI_StringAndMaybeKeyAndLength)
+	PHP_FE(sodium_crypto_generichash_init, AI_MaybeKeyAndLength)
+	PHP_FE(sodium_crypto_generichash_update, AI_StateByReferenceAndString)
+	PHP_FE(sodium_crypto_generichash_final, AI_StateByReferenceAndMaybeLength)
+#ifdef crypto_pwhash_SALTBYTES
+	PHP_FE(sodium_crypto_pwhash, AI_LengthAndPasswordAndSaltAndOpsLimitAndMemLimit)
+	PHP_FE(sodium_crypto_pwhash_str, AI_PasswordAndOpsLimitAndMemLimit)
+	PHP_FE(sodium_crypto_pwhash_str_verify, AI_HashAndPassword)
+#endif
+	PHP_FE(sodium_crypto_pwhash_scryptsalsa208sha256, AI_LengthAndPasswordAndSaltAndOpsLimitAndMemLimit)
+	PHP_FE(sodium_crypto_pwhash_scryptsalsa208sha256_str, AI_PasswordAndOpsLimitAndMemLimit)
+	PHP_FE(sodium_crypto_pwhash_scryptsalsa208sha256_str_verify, AI_HashAndPassword)
+	PHP_FE(sodium_crypto_scalarmult, AI_TwoStrings)
+	PHP_FE(sodium_crypto_secretbox, AI_StringAndNonceAndKey)
+	PHP_FE(sodium_crypto_secretbox_open, AI_StringAndNonceAndKey)
+	PHP_FE(sodium_crypto_shorthash, AI_StringAndKey)
+	PHP_FE(sodium_crypto_sign, AI_StringAndKeyPair)
+	PHP_FE(sodium_crypto_sign_detached, AI_StringAndKeyPair)
+	PHP_FE(sodium_crypto_sign_ed25519_pk_to_curve25519, AI_Key)
+	PHP_FE(sodium_crypto_sign_ed25519_sk_to_curve25519, AI_Key)
+	PHP_FE(sodium_crypto_sign_keypair, AI_None)
+	PHP_FE(sodium_crypto_sign_keypair_from_secretkey_and_publickey, AI_SecretKeyAndPublicKey)
+	PHP_FE(sodium_crypto_sign_open, AI_StringAndKeyPair)
+	PHP_FE(sodium_crypto_sign_publickey, AI_Key)
+	PHP_FE(sodium_crypto_sign_secretkey, AI_Key)
+	PHP_FE(sodium_crypto_sign_publickey_from_secretkey, AI_Key)
+	PHP_FE(sodium_crypto_sign_seed_keypair, AI_Key)
+	PHP_FE(sodium_crypto_sign_verify_detached, AI_SignatureAndStringAndKey)
+	PHP_FE(sodium_crypto_stream, AI_LengthAndNonceAndKey)
+	PHP_FE(sodium_crypto_stream_xor, AI_StringAndNonceAndKey)
+	PHP_FE(sodium_bin2hex, AI_String)
+#if SODIUM_LIBRARY_VERSION_MAJOR > 7 || \
+	(SODIUM_LIBRARY_VERSION_MAJOR == 7 && SODIUM_LIBRARY_VERSION_MINOR >= 6)
+	PHP_FE(sodium_compare, AI_TwoStrings)
+#endif
+	PHP_FE(sodium_hex2bin, AI_TwoStrings)
+	PHP_FE(sodium_increment, AI_StringRef)
+	PHP_FE(sodium_add, AI_StringRef_And_String)
+	PHP_FE(sodium_memcmp, AI_TwoStrings)
+	PHP_FE(sodium_memzero, AI_FirstArgByReferenceSecondLength)
+
+	PHP_FALIAS(sodium_crypto_scalarmult_base, sodium_crypto_box_publickey_from_secretkey, AI_TwoStrings)
+
+	PHP_FE_END
+};
+
+zend_module_entry libsodium_module_entry = {
+#if ZEND_MODULE_API_NO >= 20010901
+	STANDARD_MODULE_HEADER,
+#endif
```

Comment:
Checking for the module API here seems unnecessary, this is shipped with the latest PHP only.